code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,842 @@
+	for(var/turf/T in view(5))
+		for(var/obj/item/F in T.contents)
+			extinguishItem(F)
+		for(var/obj/machinery/light/L in T.contents)
+			L.on = 0
+			L.visible_message("<span class='warning'>[L] flickers and falls dark.</span>")
+			L.update(0)
+		for(var/obj/machinery/computer/C in T.contents)
+			C.set_light(0)
+			C.visible_message("<span class='warning'>[C] grows dim, its screen barely readable.</span>")
+		for(var/obj/structure/glowshroom/G in orange(7, user)) //High radius because glowshroom spam wrecks shadowlings
+			G.visible_message("<span class='warning'>[G] withers away!</span>")
+			qdel(G)
+		for(var/mob/living/H in T.contents)
+			extinguishMob(H)
+		for(var/mob/living/silicon/robot/borgie in T.contents)
+			borgie.update_headlamp(TRUE, 150)
+			borgie.lamp_recharging = TRUE
+			borgie.lamp_intensity = 0
+		for(var/obj/machinery/camera/cam in T.contents)
+			cam.set_light(0)
+			if(prob(10)) //small chance to completely disable a cam temporary
+				cam.emp_act(2)
+
+/obj/effect/proc_holder/spell/aoe_turf/flashfreeze //Stuns and freezes nearby people - a bit more effective than a changeling's cryosting
+	name = "Icy Veins"
+	desc = "Instantly freezes the blood of nearby people, stunning them and causing burn damage."
+	panel = "Shadowling Abilities"
+	range = 3
+	charge_max = 250
+	human_req = 1
+	clothes_req = 0
+	action_icon = 'icons/mob/actions/actions_sling.dmi'
+	action_icon_state = "icy_veins"
+	sound = 'sound/effects/ghost2.ogg'
+
+/obj/effect/proc_holder/spell/aoe_turf/flashfreeze/cast(list/targets,mob/user = usr)
+	if(!shadowling_check(user))
+		revert_cast()
+		return
+	to_chat(user, "<span class='shadowling'>You freeze the nearby air.</span>")
+	for(var/turf/T in targets)
+		for(var/mob/living/carbon/M in T.contents)
+			if(is_shadow_or_thrall(M))
+				if(M == user) //No message for the user, of course
+					continue
+				else
+					to_chat(M, "<span class='danger'>You feel a blast of paralyzingly cold air wrap around you and flow past, but you are unaffected!</span>")
+					continue
+			to_chat(M, "<span class='userdanger'>A wave of shockingly cold air engulfs you!</span>")
+			M.Stun(2)
+			M.apply_damage(10, BURN)
+			if(M.bodytemperature)
+				M.bodytemperature -= 200 //Extreme amount of initial cold
+			if(M.reagents)
+				M.reagents.add_reagent("frostoil", 15) //Half of a cryosting
+
+
+
+
+/obj/effect/proc_holder/spell/targeted/enthrall //Turns a target into the shadowling's slave. This overrides all previous loyalties
+	name = "Enthrall"
+	desc = "Allows you to enslave a conscious, non-braindead, non-catatonic human to your will. This takes some time to cast."
+	panel = "Shadowling Abilities"
+	charge_max = 0
+	human_req = 1
+	clothes_req = 0
+	range = 1 //Adjacent to user
+	action_icon_state = "enthrall"
+	action_icon = 'icons/mob/actions/actions_sling.dmi'
+	var/enthralling = 0
+
+/obj/effect/proc_holder/spell/targeted/enthrall/cast(list/targets,mob/living/carbon/human/user = usr)
```

Comment:
this entire ability should be a melee-only targetted spell, like fireball, with the mouse cursor thing.